web/src/app/users/UserContactMethodForm.tsx

```diff
@@ -176,17 +210,7 @@ export default function UserContactMethodForm(
             disabled={edit}
             component={TextField}
           >
-            {(edit || smsVoiceEnabled) && <MenuItem value='SMS'>SMS</MenuItem>}
-            {(edit || smsVoiceEnabled) && (
-              <MenuItem value='VOICE'>VOICE</MenuItem>
-            )}
-            {(edit || emailEnabled) && <MenuItem value='EMAIL'>EMAIL</MenuItem>}
-            {(edit || webhookEnabled) && (
-              <MenuItem value='WEBHOOK'>WEBHOOK</MenuItem>
-            )}
-            {(edit || slackEnabled) && (
-              <MenuItem value='SLACK_DM'>SLACK DM</MenuItem>
-            )}
+            {contactMethods.map(renderMenuItem)}
```

Comment:
This call confused me for a second. It feels like `renderMenuItem` should be called like a component, such as `<MenuItemWithDisable />`, since the function being called is essentially a functional JSX component if it was named differently

`contactMethods.map((cm) => <MenuItemWithDisable contactMethod={cm} />)` or something.

Just my 2¢

Reply:
I absolutely agree, and I tried to get that to work a lot of different ways. The tricky part is that the Form is looking for a component with the right `value` prop, and it seems like the only elegant way for it to work is for this to be a function. It didn't work as a component.

https://stackoverflow.com/questions/68378474/mui-select-custom-menuitem-not-working-properly
^^ this stackoverflow thread covers the exact issue